### Python Conditional Statements and loops

#### 1. Write a Python program to construct the following pattern, using a nested for loop. 

(double click here to see pattern)

* 
* * 
* * * 
* * * * 
* * * * * 
* * * * 
* * * 
* * 
*

In [10]:
char = "* "
for i in range(1,6):
    print(char)
    char += "* "
char = char[:-2]
for i in range(1,6):    
    char = char[:-2]
    print(char)

* 
* * 
* * * 
* * * * 
* * * * * 
* * * * 
* * * 
* * 
* 



#### 2. Write a Python program to count the number of even and odd numbers from a series of numbers. 
* Sample numbers : numbers = (1, 2, 3, 4, 5, 6, 7, 8, 9) 
##### Expected Output : 
* Number of even numbers : 5
* Number of odd numbers : 4

In [14]:
numbers = (1, 2, 3, 4, 5, 6, 7, 8, 9)
even = 0
odd  = 0
for num in numbers:
    if num & 1:            #if the LSB is 1 the number is odd
        odd += 1
    else:
        even += 1
print(f"Number of even numbers : {even}")
print(f"Number of odd numbers : {odd}")

Number of even numbers : 4
Number of odd numbers : 5


####  3. Write a Python program that prints all the numbers from 0 to 6 except 3 and 6.
* Note : Use 'continue' statement. 
* Expected Output : 0 1 2 4 5 

In [17]:
for i in range(0, 7):
    if i == 3 or i ==6:
        continue
    else:
        print(i, end=" ")

0 1 2 4 5 

#### 4. Write a Python program to get the Fibonacci series between 0 to 50. 
##### Note : The Fibonacci Sequence is the series of numbers :
* 0, 1, 1, 2, 3, 5, 8, 13, 21, .... 
* Every next number is found by adding up the two numbers before it.
* Expected Output : 1 1 2 3 5 8 13 21 34

In [32]:
def fibonacci(num):
    a, b = 0, 1
    for _ in range(num):
        a, b = b, a + b
        print(a, end=" ")

fibonacci(9)

1 1 2 3 5 8 13 21 34 

#### 5. Write a Python program which takes two digits m (row) and n (column) as input and generates a two-dimensional array. The element value in the i-th row and j-th column of the array should be i*j. 

##### Note :
* i = 0,1.., m-1 
* j = 0,1, n-1.

* Test Data : Rows = 3, Columns = 4 
* Expected Result : [[0, 0, 0, 0], [0, 1, 2, 3], [0, 2, 4, 6]]

In [36]:
import numpy as np
m = 3 #Rows
n = 4 #Columns
out = np.zeros((m,n))

for i in range(m):
    for j in range(n):
        out[i][j] = i*j
print(out)

[[0. 0. 0. 0.]
 [0. 1. 2. 3.]
 [0. 2. 4. 6.]]


#### 6. Write a Python program to find numbers between 100 and 400 (both included) where each digit of a number is an even number. The numbers obtained should be printed in a comma-separated sequence

In [40]:
start = 100
stop = 400
for i in range(start,stop+1):
    temp = str(i)
    if (int(temp[0])&1) and (int(temp[1])&1) and (int(temp[2])&1):
        print(i, end=", ")

111, 113, 115, 117, 119, 131, 133, 135, 137, 139, 151, 153, 155, 157, 159, 171, 173, 175, 177, 179, 191, 193, 195, 197, 199, 311, 313, 315, 317, 319, 331, 333, 335, 337, 339, 351, 353, 355, 357, 359, 371, 373, 375, 377, 379, 391, 393, 395, 397, 399, 

 #### 7. Write a Python program to print alphabet pattern 'A'.

#### 8. Write a Python program to print alphabet pattern 'P'

#### 9. Write a Python program to check a triangle is valid or not.

#### 10. Write a Python program to create the multiplication table (from 1 to 10) of a number.

In [45]:
i = int(input("ingrese un numero"))
print(f"Tabla del {i}")
for j in range(1,10+1):
    print(f"{i}x{j} = {i*j}")

ingrese un numero 4


Tabla del 4
4x1 = 4
4x2 = 8
4x3 = 12
4x4 = 16
4x5 = 20
4x6 = 24
4x7 = 28
4x8 = 32
4x9 = 36
4x10 = 40


#### 11. Write a Python program to construct the following pattern, using a nested loop number.
1
22
333
4444
55555
666666
7777777
88888888
999999999


In [55]:
out = ""
for i in range(1,10):
    for _ in range(i):
        out += (str(i))
        print(out)
    out = ""

1
2
22
3
33
333
4
44
444
4444
5
55
555
5555
55555
6
66
666
6666
66666
666666
7
77
777
7777
77777
777777
7777777
8
88
888
8888
88888
888888
8888888
88888888
9
99
999
9999
99999
999999
9999999
99999999
999999999
